File: Retos/Reto #1 - EL LENGUAJE HACKER [Fácil]/kotlin/mouredev.kt

fun main() {
    println(leetTranslator("Leet"))
    println(leetTranslator("Aquí está un texto de prueba para ver si funciona el reto!"))
}

private fun leetTranslator(text: String): String {

    val leet = mapOf("A" to "4", "B" to "I3", "C" to "[", "D" to ")", "E" to "3", "F" to "|=",
        "G" to "&", "H" to "#", "I" to "1", "J" to ",_|", "K" to ">|", "L" to "1", "M" to "/\\/\\",
        "N" to "^/", "O" to "0", "P" to "|*", "Q" to "(_,)", "R" to "I2", "S" to "5", "T" to "7",
        "U" to "(_)", "V" to "\\/", "W" to "\\/\\/", "X" to "><", "Y" to "j", "Z" to "2",
        "1" to "L", "2" to "R", "3" to "E", "4" to "A", "5" to "S", "6" to "b", "7" to "T",
        "8" to "B", "9" to "g", "0" to "o")

    var leetText = ""

    for (word in text) {
        val character = word.toString().uppercase()
        leetText += if (leet.containsKey(character)) {
            leet[character]
        } else {
            word.toString()
        }
    }

    return leetText
}

File: Retos/Reto #1 - EL LENGUAJE HACKER [Fácil]/python/mouredev.py
def leet_translator(text):

    leet = {"A": "4", "B": "I3", "C": "[", "D": ")", "E": "3", "F": "|=", "G": "&", "H": "#", "I": "1",
            "J": ",_|", "K": ">|", "L": "1", "M": "/\/\\", "N": " ^/", "O": "0", "P": " |*", "Q": "(_,)",
            "R": "I2", "S": "5", "T": "7", "U": "(_)", "V": "\/", "W": "\/\/", "X": "><", "Y": "j", "Z": "2",
            "1": "L", "2": "R", "3": "E", "4": "A", "5": "S", "6": "b", "7": "T", "8": "B", "9": "g", "0": "o"}

    leet_text = ""

    for word in text:
        if word.upper() in leet.keys():
            leet_text += leet[word.upper()]
        else:
            leet_text += word

    return leet_text


print(leet_translator("Leet"))
print(leet_translator("Aquí está un texto de prueba para ver si funciona el reto!"))
print(leet_translator(input("Texto a traducir: ")))
